include/util/graph_loader.hpp

```diff
@@ -103,21 +83,24 @@ inline NodeID loadNodesFromFile(std::istream &input_stream,
     barrier_node_list.shrink_to_fit();
     traffic_light_node_list.shrink_to_fit();
 
-    return n;
+    return number_of_nodes;
 }
 
 /**
  * Reads a .osrm file and produces the edges.
  */
-inline NodeID loadEdgesFromFile(std::istream &input_stream,
+inline NodeID loadEdgesFromFile(std::string &filename,
                                 std::vector<extractor::NodeBasedEdge> &edge_list)
 {
-    EdgeID m;
-    input_stream.read(reinterpret_cast<char *>(&m), sizeof(unsigned));
-    edge_list.resize(m);
-    SimpleLogger().Write() << " and " << m << " edges ";
 
-    input_stream.read((char *)edge_list.data(), m * sizeof(extractor::NodeBasedEdge));
+    storage::io::FileReader file(filename, storage::io::FileReader::VerifyFingerprint);
+    NodeID number_of_nodes = file.ReadElementCount32();
+    file.Skip<extractor::ExternalMemoryNode>(number_of_nodes);
+    EdgeID number_of_edges_filereader = file.ReadElementCount32();
```

Comment:
Also add a check after to make sure the number of edges still fits into our EdgeIDs (e.g. via a numeric_cast)